test/util_spec.js

```diff
@@ -0,0 +1,341 @@
+import {expect} from 'chai';
+import deepFreeze from 'deep-freeze';
+
+import {
+  addTagToFilter,
+  capitalize,
+  cleanseTags,
+  filterCourses,
+  filterLessons,
+  fixNonArrayTagList,
+  lessonHasTags,
+  mergeObjects,
+  removeTagFromFilter
+} from '../src/util';
+
+describe('util', () => {
+
+  describe('capitalize', () => {
+
+    it('makes first letter in a string capital', () => {
+      const text = 'a random string';
+
+      expect(capitalize(text)).to.equal('A random string');
+    });
+  });
+
+  describe('getTags', () => {
+
+  });
+
+  describe('getCourses', () => {
+
+  });
+
+  describe('mergeObjects', () => {
+
+    it('merges object by combining lists as a set of all items', () => {
+      const objA = {
+        listA: ['item1', 'item2', 'item3'],
+        listB: ['item4', 'item5'],
+        listC: []
+      };
+      const objB = {
+        listA: ['item2', 'item3', 'item4'],
+        listC: ['item6', 'item7'],
+        listD: ['item8']
+      };
+      
+      deepFreeze(objA);
+      deepFreeze(objB);
+      expect(mergeObjects(objA, objB)).to.eql({
+        listA: ['item1', 'item2', 'item3', 'item4'],
+        listB: ['item4', 'item5'],
+        listC: ['item6', 'item7'],
+        listD: ['item8']
+      });
+    });
+  });
+
+  describe('addTagToFilter', () => {
+
+    it('merges tag into filter', () => {
+      const tag = {
+        platform: ['windows']
+      };
+      const filter = {
+        platform: ['browser'],
+        subject: ['physics']
+      };
+
+      deepFreeze(tag);
+      deepFreeze(filter);
+      expect(addTagToFilter(tag, filter)).to.eql({
+        platform: ['browser', 'windows'],
+        subject: ['physics']
+      });
+    });
+  });
+
+  describe('removeTagFromFilter', () => {
```

Comment:
Denne testen er kanskje litt "snill". Litt flere "corner-cases" hadde vært fint. Enten en litt mer komplisert test, eller legge til et par flere "it()". F.eks. hva om du forsøker å fjerne en tag som ikke finnes? Hva om tag={}? Hva om tag inneholder en groupName som ikke finnes i filter? Hva om tag inneholder mer enn 1 groupName?